From 53c4e31ace1dda3aac61c4399eb026997f8f860f Mon Sep 17 00:00:00 2001
Message-Id: <53c4e31ace1dda3aac61c4399eb026997f8f860f.1413837857.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Deepak S <deepak.s@intel.com>
Date: Fri, 10 Oct 2014 21:02:46 +0530
Subject: [PATCH 311/312] FOR_UPSTREAM [VPG]: drm/i915: Enable PM interrupt
 Enable after rps disable via debugfs.

PM_IER interrupts is getting cleared when we disable rps. This patch
programs IER in "gen8_enable_rps_interrupts" function, when we enable
rps/turbo via debugfs

During enable via debugfs, we set the freq at RPe. This is help turbo behave
similarly as cherryview_enable_rps.

Issue: GMINL-2260
Change-Id: I759301d7d1a8eb59e469210df32d036994332117
Signed-off-by: Deepak S <deepak.s@intel.com>
---
 drivers/gpu/drm/i915/intel_pm.c |    8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_pm.c b/drivers/gpu/drm/i915/intel_pm.c
index 624cf3b..0751a28 100644
--- a/drivers/gpu/drm/i915/intel_pm.c
+++ b/drivers/gpu/drm/i915/intel_pm.c
@@ -3764,9 +3764,11 @@ void vlv_set_rps_mode(struct drm_device *dev, bool disable)
 			gen6_disable_rps_interrupts(dev);
 	} else {
 		I915_WRITE(GEN6_RP_CONTROL, dev_priv->rps.rps_mask);
-		if (IS_CHERRYVIEW(dev))
+		if (IS_CHERRYVIEW(dev)) {
 			gen8_enable_rps_interrupts(dev);
-		else
+			valleyview_set_rps(dev_priv->dev,
+					   dev_priv->rps.efficient_freq);
+		} else
 			gen6_enable_rps_interrupts(dev);
 	}
 }
@@ -3880,7 +3882,6 @@ void valleyview_set_rps(struct drm_device *dev, u8 val)
 		vlv_punit_write(dev_priv, PUNIT_REG_GPU_FREQ_REQ, val);
 
 	I915_WRITE(GEN6_PMINTRMSK, gen6_rps_pm_mask(dev_priv, val));
-
 	dev_priv->rps.cur_freq = val;
 	trace_intel_gpu_freq_change(vlv_gpu_freq(dev_priv, val));
 }
@@ -4037,6 +4038,7 @@ static void gen8_enable_rps_interrupts(struct drm_device *dev)
 	bdw_enable_pm_irq(dev_priv, dev_priv->pm_rps_events);
 	I915_WRITE(GEN8_GT_IIR(2), dev_priv->pm_rps_events);
 	spin_unlock_irq(&dev_priv->irq_lock);
+	I915_WRITE(GEN8_GT_IER(2), dev_priv->pm_rps_events);
 }
 
 static void gen6_enable_rps_interrupts(struct drm_device *dev)
-- 
1.7.9.5

